generative/maisi/scripts/transforms.py

```diff
@@ -0,0 +1,292 @@
+        List: A list of random intensity transforms.
+    """
+    if modality == "ct":
+        return []  # CT HU intensity is stable across different datasets
+    elif modality == "mri":
+        return [
+            RandBiasFieldd(keys=image_keys, prob=0.3, coeff_range=(0.0, 0.3)),
+            RandGibbsNoised(keys=image_keys, prob=0.3, alpha=(0.5, 1.0)),
+            RandAdjustContrastd(keys=image_keys, prob=0.3, gamma=(0.5, 2.0)),
+            RandHistogramShiftd(keys=image_keys, prob=0.05, num_control_points=10),
+        ]
+    else:
+        warnings.warn(f"No random intensity transform is applied. Modality has to be in ['ct','mri']. Got {modality}.")
+        return []
+
+
+def define_vae_transform(
+    modality: str,
+    data_aug: bool,
+    k: int = 4,
+    patch_size: List[int] = [128, 128, 128],
+    val_patch_size: Optional[List[int]] = None,
+    compute_dtype: torch.dtype = torch.float32,
+    spacing_type: str = "original",
+    spacing: Optional[List[float]] = None,
+    image_keys: List[str] = ["image"],
+    label_keys: List[str] = [],
+    additional_keys: List[str] = [],
+    select_channel: int = 0,
+) -> tuple:
+    """
+    Define the transform pipeline for training and validation.
+
+    Args:
+        modality (str): The imaging modality, either 'ct' or 'mri'.
+        data_aug (bool): Whether to apply random data augmentation.
+        k (int, optional): Patches should be divisible by k. Defaults to 4.
+        patch_size (List[int], optional): Size of the patches. Defaults to [128, 128, 128].
+        val_patch_size (Optional[List[int]], optional): Size of validation patches. Defaults to None.
+        compute_dtype (torch.dtype, optional): Final data type. Defaults to torch.float32.
+        spacing_type (str, optional): Type of spacing. Defaults to "original".
+        spacing (Optional[List[float]], optional): Spacing values. Defaults to None.
+        image_keys (List[str], optional): List of image keys. Defaults to ["image"].
+        label_keys (List[str], optional): List of label keys. Defaults to [].
+        additional_keys (List[str], optional): List of additional keys. Defaults to [].
+        select_channel (int, optional): Channel to select for multi-channel MRI. Defaults to 0.
+
+    Returns:
+        tuple: A tuple containing train_transforms and val_transforms.
+    """
+    if spacing_type not in ["original", "fixed", "rand"]:
+        raise ValueError(f"spacing_type has to be chosen from ['original', 'fixed', 'rand']. Got {spacing_type}.")
+
+    keys = image_keys + label_keys + additional_keys
+    interp_mode = ["bilinear"] * len(image_keys) + ["nearest"] * len(label_keys)
+
+    common_transform = [
+        SelectItemsd(keys=keys, allow_missing_keys=True),
+        LoadImaged(keys=keys, allow_missing_keys=True),
+        EnsureChannelFirstd(keys=keys, allow_missing_keys=True),
+        Orientationd(keys=keys, axcodes="RAS", allow_missing_keys=True),
+    ]
+
+    if modality == "mri":
```

Comment:
Can we also check if the modality input is valid? Also I would suggest that it accepts "CT", "MRI" as valid inputs as well.

Reply:
I put a warning there. It is OK to be not ct and mri, that means it uses original intensity

Reply:
Thanks!